packages/tpetra/core/src/Tpetra_Details_DistributorActor.hpp

```diff
@@ -483,6 +557,89 @@ void DistributorActor::doPosts(const DistributorPlan& plan,
   // ParameterList set by setParameterList().
   const Details::EDistributorSendType sendType = plan.getSendType();
 
+  //  All-to-all communication layout is quite different from
+  //  point-to-point, so we handle it separately.
+  if (sendType == Details::DISTRIBUTOR_ALLTOALL) {
+#ifdef HAVE_TPETRA_MPI
+    TEUCHOS_TEST_FOR_EXCEPTION(!plan.getIndicesTo().is_null(),
+                               std::runtime_error,
+                               "Send Type=\"Alltoall\" only works for fast-path communication.");
+
+    auto comm = plan.getComm();
+    std::vector<int> sendcounts (comm->getSize(), 0);
```

Comment:
Initialize to `-1` instead of `0` to increase likelihood of MPI implementation erroring